lib/core/presentation/widgets/poap/poap_item.dart

```diff
@@ -0,0 +1,300 @@
+        builder: (context, badgeDetailState) {
+          final badge = badgeDetailState.badge;
+          return FutureBuilder(
+            future: getIt<TokenRepository>().getToken(
+              input: GetTokenDetailInput(
+                id: '${badge.contract!}-0'.toLowerCase(),
+                network: badge.network,
+              ),
+            ),
+            builder: (innerContext, snapshot) {
+              final tokenDetail = snapshot.data?.fold((l) => null, (r) => r);
+              return InkWell(
+                onTap: () {
+                  BottomSheetUtils.showSnapBottomSheet(
+                    innerContext,
+                    builder: (_) => MultiBlocProvider(
+                      providers: [
+                        BlocProvider.value(
+                          value: BlocProvider.of<ClaimPoapBloc>(context),
+                        ),
+                        BlocProvider.value(
+                          value: BlocProvider.of<BadgeDetailBloc>(context),
+                        ),
+                      ],
+                      child: PopapDetailPage(
+                        tokenDetail: tokenDetail,
+                      ),
+                    ),
+                  );
+                },
+                child: Container(
+                  padding: EdgeInsets.symmetric(horizontal: Spacing.small, vertical: Spacing.small),
+                  decoration: ShapeDecoration(
+                    color: colorScheme.surfaceVariant,
+                    shape: RoundedRectangleBorder(
+                      borderRadius: BorderRadius.circular(LemonRadius.small),
+                    ),
+                  ),
+                  child: Row(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: [
+                      _PoapItemImage(tokenMetadata: tokenDetail?.metadata),
+                      SizedBox(width: Spacing.small),
+                      Flexible(
+                        child: Column(
+                          crossAxisAlignment: CrossAxisAlignment.start,
+                          children: [
+                            _PoapItemInfo(metadata: tokenDetail?.metadata),
+                            SizedBox(height: Spacing.xSmall),
+                            PoapQuantityBar(badge: badge),
+                            SizedBox(height: Spacing.small),
+                            _PoapItemButtons(badge: badge),
+                          ],
+                        ),
+                      )
+                    ],
+                  ),
+                ),
+              );
+            },
+          );
+        },
+      ),
+    );
+  }
+}
+
+class _PoapItemImage extends StatelessWidget {
+  const _PoapItemImage({
+    required this.tokenMetadata,
+  });
+
+  final TokenMetadata? tokenMetadata;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: 72,
+      height: 72,
+      decoration: ShapeDecoration(
```

Comment:
Use flutter screen utils for large raw size. Ex 72.w

Reply:
👍